client/src/main.rs

```diff
@@ -59,9 +59,14 @@ struct Opts {
     #[clap(short, long, default_value = "/etc/innernet")]
     config_dir: PathBuf,
 
+    #[cfg(not(target_os = "openbsd"))]
     #[clap(short, long, default_value = "/var/lib/innernet")]
     data_dir: PathBuf,
 
+    #[cfg(target_os = "openbsd")]
+    #[clap(short, long, default_value = "/var/db/innernet")]
+    data_dir: PathBuf,
```

Comment:
It should be possible to do this with slightly less duplication:
```suggestion
    #[cfg_attr(not(target_os = "openbsd"), clap(short, long, default_value = "/var/lib/innernet"))]
    #[cfg_attr(target_os = "openbsd", clap(short, long, default_value = "/var/db/innernet"))]
    data_dir: PathBuf,
```

Or it may be possible to use something like
```rust
#[cfg(not(target_os = "openbsd"))]
const DEFAULT_DATA_DIR: &'static str = "/var/lib/innernet";
#[cfg(target_os = "openbsd")]
const DEFAULT_DATA_DIR: &'static str = "/var/db/innernet";
```
...and use the constant in the clap attribute. The the latter works I think it would be preferred.

Same for server main.rs.

Reply:
Much nicer, thanks. The second option doesn't work, seems like taking a `const` as a value for an attribute is not supported, unless I'm missing something.